apps/gateway/package.json

```diff
@@ -22,6 +22,7 @@
     "@ai-sdk/cohere": "^2.0.19",
     "@ai-sdk/google-vertex": "^3.0.51",
     "@ai-sdk/groq": "^2.0.28",
+    "@ai-sdk/provider": "^3.0.0",
```

Comment:
Version 3 of @ai-sdk/provider (and with that LanguageModelV3...) was published with AI-SDK v6. 

Since you are requiring this, you also need to
1/ Bump version of ai to ^6
2/ Check whether there is anything else we need to migrate:

https://vercel.com/blog/ai-sdk-6

Reply:
The reason for adding `@ai-sdk/provider` is about reusing `LanguageModelV3Prompt` for type hint in typescript, I could just lower it to `LanguageModelV2Prompt` (available in current package too) to limit scope of this PR rather than upgrade AI SDK to v6 (which makes better sense to be done in separate PR).

Reply:
According to the release notes, AI SDK 6 has minimal breaking changes. Suggest to spend 10-15 minutes checking whether there is anything breaking / deprecated that we need to migrate. If minimal, do it as part of this PR. Otherwise depend on v2 of @ai-sdk/provider.